extern/storage-sealing/precommit_policy.go

```diff
@@ -79,11 +83,41 @@ func (p *BasicPreCommitPolicy) Expiration(ctx context.Context, ps ...Piece) (abi
 	}
 
 	if end == nil {
-		tmp := epoch + p.duration
+		// no deal pieces, get expiration for committed capacity sector
+		expirationDuration, err := p.getCCSectorLifetime()
+		if err != nil {
+			return 0, err
+		}
+
+		tmp := epoch + expirationDuration
 		end = &tmp
 	}
 
 	*end += miner.WPoStProvingPeriod - (*end % miner.WPoStProvingPeriod) + p.provingBoundary - 1
 
 	return *end, nil
 }
+
+func (p *BasicPreCommitPolicy) getCCSectorLifetime() (abi.ChainEpoch, error) {
+	c, err := p.getSealingConfig()
+	if err != nil {
+		return 0, xerrors.Errorf("sealing config load error: %w", err)
+	}
+
+	var ccLifetimeEpochs = abi.ChainEpoch(uint64(c.CommittedCapacitySectorLifetime.Truncate(builtin.EpochDurationSeconds).Seconds()) / builtin.EpochDurationSeconds)
+	// if zero value in config, assume maximum sector extension
+	if ccLifetimeEpochs == 0 {
+		ccLifetimeEpochs = policy.GetMaxSectorExpirationExtension()
+	}
+
+	if minExpiration := abi.ChainEpoch(miner.MinSectorExpiration); ccLifetimeEpochs < minExpiration {
```

Comment:
I tried to put this into the `policy` package, but it appears to be autogen'd code. LMK if you'd like for me to abstract this upstream.

Reply:
`policy.go` is generated form  `policy.go.template`, so you'd just put this in that file, and run `make actors-gen`

(I'm also not sure if it's worth putting this code there since it doesn't do anything actor-version specific, and likely won't be used outside of the sealing package)